packages/ai/src/types/language-model.ts

```diff
@@ -6,10 +7,58 @@ import {
   LanguageModelV3Source,
 } from '@ai-sdk/provider';
 
+declare global {
+  /**
+   * Global interface that can be augmented by third-party packages to register custom model IDs.
+   *
+   * You can register model IDs in two ways:
+   *
+   * 1. Register baesd on Model IDs from a provider package:
+   * @example
+   * ```typescript
+   * import { openai } from '@ai-sdk/openai';
+   * type OpenAIResponsesModelId = Parameters<typeof openai>[0];
+   *
+   * declare global {
+   *   interface RegisteredProviderModels {
+   *     openai: OpenAIResponsesModelId;
+   *   }
+   * }
+   * ```
+   *
+   * 2. Register individual model IDs directly as keys:
+   * @example
+   * ```typescript
+   * declare global {
+   *   interface RegisteredProviderModels {
+   *     'my-provider:my-model': any;
+   *     'my-provider:another-model': any;
+   *   }
+   * }
+   * ```
+   */
+  interface RegisteredProviderModels {}
+}
```

Comment:
@nicoalbanese let me know about the DX. I think we could document it in https://ai-sdk.dev/docs/ai-sdk-core/provider-management#global-provider-configuration as well, and could help developers of other gateways to register their own IDs that way

https://github.com/user-attachments/assets/c2cf9aaa-5a60-4d58-b592-80679f052fde